index.js

```diff
@@ -1,66 +1,190 @@
-module.exports = {
-    "extends": ["airbnb-base"],
-    "plugins": [
-        "import"
-    ],
-    "env": {
-        "node": true,
-        "browser": true
+const globals = require('globals');
+const js = require('@eslint/js');
+const typescriptEslint = require('typescript-eslint');
+
+const {
+    FlatCompat,
+} = require('@eslint/eslintrc');
+
+const compat = new FlatCompat({
+    baseDirectory: __dirname,
+    recommendedConfig: js.configs.recommended,
+    allConfig: js.configs.all
+});
+
+module.exports = [...compat.extends('airbnb-base'),
+    {
+        linterOptions: {
+            reportUnusedDisableDirectives: true,
+        },
+
+        languageOptions: {
+            globals: {
+                ...globals.node,
+                ...globals.browser,
+            },
+
+            ecmaVersion: 2022,
+        },
+
+        rules: {
+            indent: ['error', 4, {
+                SwitchCase: 1,
+            }],
+            // Allow underscore dangle for a few specific properties, mainly `_id`.
+            'no-underscore-dangle': [2, {
+                allow: [
+                    '_id',
+                    '_outputSeqNo',
+                    '_queueOrderNo',
+                    '_skipOutput',
+                    '_retryCount',
+                    '_crashesCount',
+                    '_pageFailed',
+                    '_pageCrashed',
+                ],
+                allowAfterThis: true,
+            }],
+            // ++ and -- are fine.
+            'no-plusplus': 'off',
+            quotes: ['error', 'single', {
+                avoidEscape: true,
+                allowTemplateLiterals: true,
+            }],
+            // Disable as forcing names for function expression doesn't bring much value.
+            'func-names': 'off',
+            // Disable as this rule is overly strict and can cause breaking changes.
+            'class-methods-use-this': 'off',
+            // Forcing "use strict" is not necessary.
+            strict: 'off',
+            // We don't want to force newlines before and after curly braces.
+            'object-curly-newline': 'off',
```

Comment:
I would not completely disable this, but force consistency (either no newlines at all, or before/after both braces), what do you think?
```suggestion
            'object-curly-newline': ['error', { 'consistent': true }],
```

Reply:
Makes sense, thanks